stk/calculators/optimization/optimizers.py

```diff
@@ -662,88 +672,457 @@ class GFNXTB(Optimizer):
+                raise XTBInvalidSolventError(
+                    f'No solvent valid for version',
+                    f' {gfn_version!r}.'
+                )
+            if not is_valid_xtb_solvent(gfn_version, solvent):
+                raise XTBInvalidSolventError(
+                    f'Solvent {solvent!r} is invalid for ',
+                    f'version {gfn_version!r}.'
+                )
 
-        self.gfnxtb_path = gfnxtb_path
+        self.xtb_path = xtb_path
+        self.gfn_version = str(gfn_version)
         self.output_dir = output_dir
+        self.opt_level = opt_level
+        self.max_runs = max_runs
         self.num_cores = str(num_cores)
+        self.electronic_temperature = str(electronic_temperature)
+        self.solvent = solvent
+        self.solvent_grid = solvent_grid
+        self.charge = str(charge)
+        self.num_unpaired_electrons = str(num_unpaired_electrons)
+        self.unlimited_memory = unlimited_memory
+        self.incomplete = set()
         super().__init__(use_cache=use_cache)
 
+    def _check_neg_frequencies(self, output_file):
+        """
+        Check for negative frequencies.
+
+        Parameters
+        ----------
+        output_file : :class:`str`
+            Name of output file with xTB results.
+
+        Returns
+        -------
+        :class:`bool`
+            Returns `True` if a negative frequency is present
+
+        """
+        neg_freq = False
+        xtbext = XTBExtractor(output_file=output_file)
+        value = xtbext.frequencies()
+        # Check for one negative frequency, excluding the first
+        # 6 frequencies.
+        if min(value[7:]) < 0:
+            neg_freq = True
+        return neg_freq
+
+    def _incomplete(self, output_file):
+        """
+        Check if xTB optimization has converged.
+
+        Parameters
+        ----------
+        output_file : :class:`str`
+            Name of output file with xTB results.
+
+        Returns
+        -------
+        :class:`bool`
+            Returns `True` if a negative frequency is present. Raises
+            errors if optimization did not converge.
+
+        """
+        if output_file is None:
+            # No simulation has been run.
+            return True
+        # If convergence is achieved, then .xtboptok should exist.
+        if os.path.exists('.xtboptok'):
+            # Check for negative frequencies in output file in max_run
+            # is not None..
+            # Return True if there exists at least one.
+            if self.max_runs is not None:
+                return self._check_neg_frequencies(output_file)
+            else:
+                return False
+        elif os.path.exists('NOT_CONVERGED'):
+            XTBConvergenceError('Optimization not converged.')
+        else:
+            XTBOptimizerError('Optimization failed to complete')
+
+    def _run_xtb(self, xyz, out_file):
+        """
+        Runs the command for GFN-xTB using subprocess.
```

Comment:
The use of subprocess is an implementation detail and should not be mentioned in the docstring.